File: src/main/java/upbrella/be/store/dto/response/AllCurrentLocationStoreResponse.java
package upbrella.be.store.dto.response;

import lombok.Builder;
import lombok.Getter;

import java.util.List;

@Getter
@Builder
public class AllCurrentLocationStoreResponse {

    private List<SingleCurrentLocationStoreResponse> stores;

    public static AllCurrentLocationStoreResponse ofCreate(List<SingleCurrentLocationStoreResponse> singleCurrentLocationStoreResponses) {

        return AllCurrentLocationStoreResponse.builder()
                .stores(singleCurrentLocationStoreResponses)
                .build();
    }
}

File: src/main/kotlin/upbrella/be/store/dto/response/AllCurrentLocationStoreResponse.kt
package upbrella.be.store.dto.response

data class AllCurrentLocationStoreResponse(
    val stores: List<SingleCurrentLocationStoreResponse>
) {
    companion object {
        fun ofCreate(singleCurrentLocationStoreResponses: List<SingleCurrentLocationStoreResponse>) =
            AllCurrentLocationStoreResponse(singleCurrentLocationStoreResponses)
    }
}
